scripts/lib/types.ts

```diff
@@ -0,0 +1,120 @@
+import * as admin from "firebase-admin";
+
+// re-export type definitions to decrease declaration verbosity in other files
+// and for some reason, ESLint thinks they're unused, thus the disable comments
+//
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export import CollectionReference = admin.firestore.CollectionReference;
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export import DocumentData = admin.firestore.DocumentData;
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export import DocumentReference = admin.firestore.DocumentReference;
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export import DocumentSnapshot = admin.firestore.DocumentSnapshot;
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export import QueryDocumentSnapshot = admin.firestore.QueryDocumentSnapshot;
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export import Timestamp = admin.firestore.Timestamp;
+
+export type LogFunction = typeof console.log;
+
+export type Incomplete<T> = {
+  [K in keyof T]?: T[K] extends object ? Incomplete<T[K]> : T[K];
+};
```

Comment:
`Incomplete` a.k.a. Recursive `Partial` type. This one allows to define a JSON parse result with the possibility of any of the fields being missing.